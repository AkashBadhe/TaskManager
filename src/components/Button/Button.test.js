import React from "react";
import { shallow } from "enzyme";
import Button from "./index";

describe("Button", () => {
  let button = shallow(<Button text={"Add"} />);
  it("shoud mount the Button", () => {
    expect(button).toMatchSnapshot();
  });
  it("should render Add button", () => {
    let addButton = shallow(<Button text={"Add"} variant="add" />);
    expect(addButton.props().variant).toBe("add");
    expect(addButton.text()).toBe("Add");
  });
});
