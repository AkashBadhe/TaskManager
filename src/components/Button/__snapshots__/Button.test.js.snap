// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button shoud mount the Button 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    text="Add"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "Add",
    },
    "ref": null,
    "rendered": "Add",
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "baseHash": 400283751,
        "componentId": "sc-AxjAm",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  justify-content: center;
  align-items: center;
  color: rgb(46, 68, 78);
  line-height: 18px;
  border: none;
  transition-duration: 0.085s;
  padding: 4px 8px;
  cursor: pointer;
  border-radius: ",
          [Function],
          ";
  font-weight: ",
          [Function],
          ";
  font-size: ",
          [Function],
          ";
  margin: ",
          [Function],
          ";
  width: ",
          [Function],
          ";
  height: ",
          [Function],
          ";
  transition-timing-function: ease-in;
  background-color: ",
          [Function],
          ";

  &:hover,
  &:focus,
  &:active {
    box-shadow: inset 0 0 0 2px rgba(0, 0, 0, 0.2);
  }
  &:disabled {
    cursor: not-allowed;
    box-shadow: none;
  }
",
        ],
        "staticRulesId": "",
      },
      "displayName": "styled.button",
      "foldedComponentIds": Array [],
      "render": [Function],
      "shouldForwardProp": undefined,
      "styledComponentId": "sc-AxjAm",
      "target": "button",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "Add",
      },
      "ref": null,
      "rendered": "Add",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "baseHash": 400283751,
          "componentId": "sc-AxjAm",
          "isStatic": false,
          "rules": Array [
            "
  display: flex;
  justify-content: center;
  align-items: center;
  color: rgb(46, 68, 78);
  line-height: 18px;
  border: none;
  transition-duration: 0.085s;
  padding: 4px 8px;
  cursor: pointer;
  border-radius: ",
            [Function],
            ";
  font-weight: ",
            [Function],
            ";
  font-size: ",
            [Function],
            ";
  margin: ",
            [Function],
            ";
  width: ",
            [Function],
            ";
  height: ",
            [Function],
            ";
  transition-timing-function: ease-in;
  background-color: ",
            [Function],
            ";

  &:hover,
  &:focus,
  &:active {
    box-shadow: inset 0 0 0 2px rgba(0, 0, 0, 0.2);
  }
  &:disabled {
    cursor: not-allowed;
    box-shadow: none;
  }
",
          ],
          "staticRulesId": "",
        },
        "displayName": "styled.button",
        "foldedComponentIds": Array [],
        "render": [Function],
        "shouldForwardProp": undefined,
        "styledComponentId": "sc-AxjAm",
        "target": "button",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
